apis/refs/v1beta1/computerefs.go

```diff
@@ -76,44 +59,54 @@ func ResolveComputeNetwork(ctx context.Context, reader client.Reader, src client
 		key.Namespace = src.GetNamespace()
 	}
 
-	computenetwork := &unstructured.Unstructured{}
-	computenetwork.SetGroupVersionKind(schema.GroupVersionKind{
+	computeNetwork := &unstructured.Unstructured{}
+	computeNetwork.SetGroupVersionKind(schema.GroupVersionKind{
 		Group:   "compute.cnrm.cloud.google.com",
 		Version: "v1beta1",
 		Kind:    "ComputeNetwork",
 	})
-	if err := reader.Get(ctx, key, computenetwork); err != nil {
+	if err := reader.Get(ctx, key, computeNetwork); err != nil {
 		if apierrors.IsNotFound(err) {
-			return nil, k8s.NewReferenceNotFoundError(computenetwork.GroupVersionKind(), key)
+			return nil, k8s.NewReferenceNotFoundError(computeNetwork.GroupVersionKind(), key)
 		}
 		return nil, fmt.Errorf("error reading referenced ComputeNetwork %v: %w", key, err)
 	}
 
-	computenetworkID, err := GetResourceID(computenetwork)
+	resourceID, err := GetResourceID(computeNetwork)
 	if err != nil {
 		return nil, err
 	}
 
-	computeNetworkProjectID, err := ResolveProjectID(ctx, reader, computenetwork)
+	projectID, err := ResolveProjectID(ctx, reader, computeNetwork)
 	if err != nil {
 		return nil, err
 	}
-	return &ComputeNetwork{
-		Project:          computeNetworkProjectID,
-		ComputeNetworkID: computenetworkID,
-	}, nil
+
+	// Need to determine good targetField values, some common values: selfLink, name, id, email, or specific field name etc
```

Comment:
Can we make this methods instead e.g. SelfLink() and ID().  Then the compiler is working with us.